### ISSUE TITLE
0.18.1 Cannot automatically detect .app

### ISSUE THREAD

**Issue by jmoody** [calabash/calabash-ios#1008]
```
1. Collect all the app bundles in the project DerivedData directory.
2. Check if the app bundle is a valid .app (built for the simulator and has the correct directory structure)
3. Check to see if the app is linked with calabash.
4. Sort the remaining by `File.mtime` time and chose the most recent.
```

* Step 1 is broken - we are not detecting the correct DerivedData directory (out of the many that might be present)
* Step 4 is broken - `File.mtime` + max (or sort) is not working as expected

ATTN @ark-konopacki 

Thanks @topgenorth for helping me debug.

**Comment by jmoody** [comment]
ATTN @Emibob, @zeljkoilic, @arulami

I am @mention'ing you because you have all had problems with Calabash auto detecting the .app.

I was able to do a remote debug session with a user who was having the same problem: Calabash cannot auto-detect the .app.

I was able to identify two problems in the algorithm and this morning I think I found a solution.  The problem is that I cannot reproduce this problem locally.  Is there any chance I could do a remote debugging session on your machine?

I would want to use https://tmate.io/

We can text chat on [Gitter](https://gitter.im/calabash/calabash0x?utm_source=share-link&utm_medium=link&utm_campaign=share-link) or any where else.

In a nutshell, I want to ssh to your machine and do some debugging in your environment.  You can be present or not.  I just need an environment where the problem is happening.

Thanks!

**Comment by Emibob** [comment]
@jmoody unfortunately I don't think my employer will allow me to give you access :( hopefully one of the other developers you mentioned will be able to

**Comment by carlocepres** [comment]
I experiencing the same issue with this problem for Unable to auto detect a .app that is linked Calabash. i was able to run my cal target on xcode but when I run it on my directory on the terminal the error is: 

 Unable to auto detect a .app that is linked Calabash.
 
 Searched: /Users/FDMCarlo/Library/Developer/Xcode/DerivedData/Build/Products
 
 Have you built your app for simulator?
 
 Please build your app from Xcode
 
 Alternatively, specify APP in features/support/01_launch.rb
 or as an environment variable:
 
 $ APP=/path/to/Your.app bundle exec cucumber
  (RuntimeError)
 ./features/support/01_launch.rb:29:in `Before'

**Comment by jmoody** [comment]
Can those of you who are having a problem, please try the following.

The instructions assume you are using bundler.  If you are not using bundler, please triple check that you are using run-loop 2.0.9 before reporting problems. 
```
1. Build your app targetting an iOS Simulator from Xcode or scripts

2. In the directory where you run cucumber
$ bundle update

3. Verify that you are using run-loop 2.0.9
$ bundle show run_loop

4. Open a console and make sure that APP and APP_BUNDLE_PATH are empty
$ APP= APP_BUNDLE_PATH= be calabash-ios console

5. See if we can find your app
> RunLoop::DetectAUT::Detect.new.app_for_simulator
```

I have not wired this new RunLoop behavior up to Calabash, so don't expect that Calabash can find your app.

Assuming all goes well, it will be available in 0.18.2.

**Comment by Emibob** [comment]
@jmoody this is working for me now after updating to 18.2 & moving my built app from Derived Data to the spec directory

**Comment by Emibob** [comment]
Thank you!

**Comment by jmoody** [comment]
@Emibob 

Calabash should have been able to find the .app in your DerivedData directory.  Can you try that?

**Comment by Emibob** [comment]
@jmoody I tried that & it was not able to find it in DerivedData - I got the message that said to make sure your .app is in the current directory or a child directory & DerivedData was not, so I moved the built file inside my spec folder & it was able to find all files & tests passed.

**Comment by jmoody** [comment]
The message should have listed several directories that were searched.

Can you show me the whole error?  Please redact the names if necessary.  Or you can send me the error in a private message or on [Gitter](https://gitter.im/calabash/calabash0x?utm_source=share-link&utm_medium=link&utm_campaign=share-link)

**Comment by Emibob** [comment]
@jmoody here's the error message when the app is in DerivedData: 
![screen shot 2016-03-11 at 2 35 06 pm](https://cloud.githubusercontent.com/assets/4623303/13713365/9f68466a-e796-11e5-9c73-b90cd24cc999.png)

**Comment by jmoody** [comment]
@Emibob 

Do you have an Xcode project?  If so, try setting XCODEPROJ

```
# Assuming your .xcodeproj is located in ../ and named ThisAM.xcodeproj
$ XCODEPROJ=../ThisAM.xcodeproj DEVICE_TARTET= etc. cucumber
```

**Comment by Emibob** [comment]
@jmoody yes that works too! Thank you :dancer:

**Comment by arulami** [comment]
@jmoody Is this need to fix additionally? why its not able to find .xcodeproj? any idea?

**Comment by jmoody** [comment]
@arulami In @Emibob case, the `xcodeproj` is located _above_ the directory where `cucumber` is run.  The algorithm only searches the current directory and below.

I think searching one directory above _might_ be OK, but it also might lead to false positives.  I am interested in any and all feedback around this.

**Comment by arulami** [comment]
@jmoody: Searching the xcodeproj in current directory and below would be fine. 
 Also, It would be easy for users to understand, if error message(RunLoop::NoSimulatorAppFoundError) directs to set Environment variable, if search fails.

> XCODEPROJ=../ThisAM.xcodeproj DEVICE_TARGET= etc. cucumber

Thanks for making me understanding.
